src/main/java/interface_adaptors/gatewayReader.java

```diff
@@ -0,0 +1,31 @@
+package interface_adaptors;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+public class gatewayReader {
+
+    private final File file;
+
+    public gatewayReader(String filePath){
+        this.file = new File(filePath);
+    }
+
+    public gatewayReader(File file){
+        this.file = file;
```

Comment:
I am not quite sure what this means. Are we supposed to assume we have a file type instance attribute stored in this class?